### ISSUE TITLE
customParseFormat plugin pars invalid date

### ISSUE THREAD

**Issue by FBerthelot** [iamkun/dayjs#737]
Hello,

When trying to parse some invalid date like `"54/42/0000 00:00", "MM/DD/YYYY HH:mm"`, dayjs doesn't set the date as invalid and set a "random" value instead. 

You can test here with a comparaison with moment [https://codesandbox.io/s/eager-frost-x1ng0](https://codesandbox.io/s/eager-frost-x1ng0).

Thank you :)

**Comment by BePo65** [comment]
These are not 'random values' :-) In both cases dayjs is different from moment.

1. in case of `00/00/0000` the current date is used
2. in case of `54/42/0000` the year is replaced by the currrent year and the day and the month create 'overflows'; this means from 'January 32th' dayjs creates 'February 1st' etc. Hence your example creates a date of '07/12/2026' (when running on '2022-06-02')

**Comment by FBerthelot** [comment]
Hey thanks @BePo65  for the answer !

If it's not random values, I think it will be valuable to add it to the documentation.
What do you think ?

**Comment by BePo65** [comment]
> If it's not random values, I think it will be valuable to add it to the documentation.

You are right, that should be part of the documentation. In my [pr 1914](https://github.com/iamkun/dayjs/pull/1914#issuecomment-1139888043) I proposed such a segment in the documentation.

Perhaps it would be a good idea to make a separate pr out of this  topic.

**Comment by FBerthelot** [comment]
Awesome, you made a PR with it !

Let's wait the merge of your PR then. 
As soon as your PR is merge, I will close this issue.

Thanks for #1914  ;)
